app/javascript/packages/document-capture/components/acuant-capture.tsx

```diff
@@ -353,20 +357,15 @@ function AcuantCapture(
     onChangeAndResetError(nextValue, analyticsPayload);
   }
 
+  type LoggedClickCallback = (...args: any[]) => any;
+
   /**
    * Given a click source, returns a higher-order function that, when called, will log an event
    * before calling the original function.
-   *
-   * @template {(...args: any[]) => any} T
-   *
-   * @param {string} source Click source.
-   * @param {{isDrop: boolean}=} metadata Additional payload metadata to log.
-   *
-   * @return {(fn: T) => (...args: Parameters<T>) => ReturnType<T>}
    */
-  function withLoggedClick(source, metadata = { isDrop: false }) {
-    return (fn) =>
-      (...args) => {
+  function withLoggedClick(source: string, metadata: { isDrop: boolean } = { isDrop: false }) {
+    return (fn: LoggedClickCallback) =>
+      (...args: Parameters<LoggedClickCallback>) => {
```

Comment:
We could do similar here with retaining the generic type:

```suggestion
    return <T extends (...args: any[]) => any>(fn: T) =>
      (...args: Parameters<T>) => {
```